python.cc

```diff
@@ -1,4 +1,134 @@
 #include "libpstack/python.h"
+#include <dlfcn.h>
+#include <string.h>
+#include <regex.h>
+
+static
+std::tuple<Elf::Object::sptr, Elf::Addr, Elf::Addr>
+getInterpHead(const Process &);
+
+template <int V>
+std::tuple<Elf::Object::sptr, Elf::Addr, Elf::Addr>
+getInterpHead(const Process &);
+
+PyInterpInfo
+getPyInterpInfo(const Process &proc) {
+    Elf::Object::sptr libpython;
+    Elf::Addr libpythonAddr;
+    Elf::Addr interpreterHead;
+    
+    std::tie(libpython, libpythonAddr, interpreterHead) = getInterpHead(proc);
+
+    std::string filename = libpython->io->filename();
+
+    auto index = filename.find("python");
+
+    if (filename.length() < index + 9) //index + len("pythonX.Y")
+        throw Exception() << "Can't parse python version from lib/exec name: " << filename;
+
+    char majorChar = filename[index + 6];
+    char minorChar = filename[index + 8];
+
+    if (!isdigit(majorChar) || !isdigit(minorChar))
+        throw Exception() << "lib/exec name doesn't match \"*pythonX.Y.*\" format";
+
+    int major = majorChar - '0';
+    int minor = minorChar - '0';
+
+    if (verbose)
+        std::clog << "python version is: " << major << "." << minor << std::endl;
+
+    return PyInterpInfo {
+        libpython, libpythonAddr, interpreterHead, 
+        "v" + std::to_string(major) + "." + std::to_string(minor),
+        V2HEX(major, minor)};
+}
+
+template <>
```

Comment:
You should put the version 2 python files in python2.cc and the version 3 files in python3.cc, otherwise you'll end up with conflicts if you need to use content in the python headers when compiling.